packages/core/src/compiler/Substance.test.ts

```diff
@@ -277,12 +277,93 @@ NoLabel B, C
     const env = envOrError(domainProg);
     subEnvOrError(prog, env);
   });
+  describe("indexed expressions", () => {
+    test("indexed set decl - sets with even indices", () => {
+      const env = envOrError(domainProg);
+      const prog = `Set a_i for i in [1, 10] where i % 2 == 0`;
+      const res = compileSubstance(prog, env);
+      expect(res.isOk()).toBe(true);
+      if (res.isOk()) {
+        hasVars(res.value[1], [
+          ["a_2", "Set"],
+          ["a_4", "Set"],
+          ["a_6", "Set"],
+          ["a_8", "Set"],
+          ["a_10", "Set"],
+        ]);
+      }
+    });
+
+    test("indexed set decl - no satisfying indices", () => {
+      const env1 = envOrError(domainProg);
+      const prog1 = `Set a_i for i in [1, 10] where i % 2 == 0.5`;
+      const res1 = compileSubstance(prog1, env1);
+      expect(res1.isOk()).toBe(true);
+      if (res1.isOk()) {
+        expect(res1.value[1].vars.size).toBe(0);
+      }
+
+      const env2 = envOrError(domainProg);
+      const prog2 = `Set a_i for i in [11, 10]`;
```

Comment:
This is because the index ranges need to be ascending?

Reply:
Yes. My interpretation is that `i in [11, 10]` means `i` is an integer and `i >= 11` and `i <= 10` which is impossible.

Reply:
got it. can you add a quick thing to the docs on this?

Reply:
sounds good, will do it